server/plugins/org.eclipse.glsp.ide.editor/src/org/eclipse/glsp/ide/editor/internal/actions/ModifyCSSFeedbackOperation.java

```diff
@@ -0,0 +1,52 @@
+/********************************************************************************
+ * Copyright (c) 2021 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.ide.editor.internal.actions;
+
+import java.util.List;
+
+import org.eclipse.glsp.server.operations.Operation;
+
+public class ModifyCSSFeedbackOperation extends Operation {
```

Comment:
I'm a litte bit confused why this is typed as `Operation`. Since this is client side only feedback and is not part of our Server-Operation API it should be typed as plain `Action`. It is also typed as `Action` on the client side.

Also all properties of this action are in theory optional (At least that's they are defined on the client side).
Maybe we should indicate this by returning an Optional in the corresponding getter-Method?

Reply:
You are right. I assumed it should be an operation because it modifies the CSS classes but it's better to have it as it is on the client side.

I'm alright with returning Optionals for optional parameters, will update all actions accordingly.